### PR TITLE
change FusionReshapeReductionShmoo from test_f to test_p

### PR DESCRIPTION
**Pull request by liqiangxl**
Issue: #2463 
This is the 1st PR to change shmoo tests from test_f to test_p
namespace is used to avoid potentail name conflicts e.g. when have multiple suites using same named para structures and functions to generate paras.

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build

File: tests/cpp/test_gpu_view.cpp [hunk 1]
```diff
@@ -353,38 +353,49 @@ std::vector<reshape_example> all_reshape_examples = {
...
-      {{1, 333, 1}, {333}},
-      {{1, 27454, 1, 2}, {1, 3922, 1, 7}},
-      {{1, 7844, 1, 7}, {1, 1961, 4}}};
-
-  for (auto e : reshape_after_reduce_examples) {
-    maybeClearAllocator(); // see above
-    reductionViewAddFusion(
-        e.first, e.second, false /* reshape_before_reduction */);
-  }
+std::vector<reshape_example> reshape_after_reduce_examples = {
```
Comment (wujingyue) [inline tests/cpp/test_gpu_view.cpp:356]: I understand it's not your doing, but would you mind changing reshape_example to `using ReshapeExample = std::pair<shape_t, shape_t>;`? This keeps the coding style consistent.

File: tests/cpp/test_gpu_view.cpp [hunk 2]
```diff
@@ -353,38 +353,49 @@ std::vector<reshape_example> all_reshape_examples = {
...
+    {{1, 333, 1}, {1, 37, 9, 1}},
+    {{22, 1, 22, 1}, {484}},
+    {{1, 333, 1}, {333}},
+    {{1, 27454, 1, 2}, {1, 3922, 1, 7}},
+    {{1, 7844, 1, 7}, {1, 1961, 4}}};
+
+namespace {
+using ReshapeBeforeReduction = NVFuserFixtureParamTest<reshape_example>;
+TEST_P(ReshapeBeforeReduction, FusionReshapeBeforeReduction) {
+  auto e = GetParam();
```
Comment (wujingyue) [inline tests/cpp/test_gpu_view.cpp:375]: I think structured binding would work better. 

```suggestion
  const auto& [input_shape, output_shape] = GetParam();
```

It gives names to `e.first` and `e.second`.

File: tests/cpp/test_gpu_view.cpp [hunk 3]
```diff
@@ -353,38 +353,49 @@ std::vector<reshape_example> all_reshape_examples = {
...
+  maybeClearAllocator(); // Shmoo tests can occupy a lot of memory
+  reductionViewAddFusion(
+      e.first, e.second, true /* reshape_before_reduction */);
```
Comment (wujingyue) [inline tests/cpp/test_gpu_view.cpp:378]:
```suggestion
      e.first, e.second, /*reshape_before_reduction=*/true);
```
to enforce the clang-tidy check: https://clang.llvm.org/extra/clang-tidy/checks/bugprone/argument-comment.html#bugprone-argument-comment

File: tests/cpp/test_gpu_view.cpp [hunk 4]
```diff
@@ -353,38 +353,49 @@ std::vector<reshape_example> all_reshape_examples = {
...
+INSTANTIATE_TEST_SUITE_P(
+    ,
+    ReshapeBeforeReduction,
+    ::testing::ValuesIn(all_reshape_examples));
+} // namespace
+
+namespace {
+using ReshapeAfterReduction = NVFuserFixtureParamTest<reshape_example>;
+TEST_P(ReshapeAfterReduction, FusionReshapeAfterReduction) {
+  auto e = GetParam();
```
Comment (wujingyue) [inline tests/cpp/test_gpu_view.cpp:389]: ditto

**Review by wujingyue — APPROVED** [review]

File: tests/cpp/test_gpu_view.cpp (hunk 1 above)
Comment (liqiangxl) [reply tests/cpp/test_gpu_view.cpp]: Good point! changed.

File: tests/cpp/test_gpu_view.cpp (hunk 4 above)
Comment (liqiangxl) [reply tests/cpp/test_gpu_view.cpp]: revised.

**Pull request merged**